### PR TITLE
More flexible

### PR DESCRIPTION
**Pull request by andresgomezfrr**
# TL;DR
Add support to create nodes using a transform `SdkTransform[InputT, OutputT]` and an input `Map[String, SdkBindingData[_]]`.

This was supported previously and we removed this feature with the strongly-types interface.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

### CODE REVIEW

File: flytekit-java/src/test/java/org/flyte/flytekit/SdkWorkflowBuilderTest.java
Comment (narape) [inline flytekit-java/src/test/java/org/flyte/flytekit/SdkWorkflowBuilderTest.java:1]: Add a test where you pass a map with more inputs than needed and a second one where you pass a map with fewer inputs than needed by the tasks

File: flytekit-scala_2.13/src/main/scala/org/flyte/flytekitscala/SdkScalaWorkflow.scala
Comment (narape) [inline flytekit-scala_2.13/src/main/scala/org/flyte/flytekitscala/SdkScalaWorkflow.scala:1]: I just realized that we don't have tests for this class

**Review by narape — COMMENTED** [review]
A good way to recover some dynamic typing

File: flytekit-scala_2.13/src/main/scala/org/flyte/flytekitscala/SdkScalaWorkflow.scala
Comment (narape) [reply flytekit-scala_2.13/src/main/scala/org/flyte/flytekitscala/SdkScalaWorkflow.scala:1]: There is very little code in this class. We can disregard my comment

**Review by narape — APPROVED** [review]
